lib/ssh_scan/config_loader/yaml.rb

```diff
@@ -0,0 +1,53 @@
+require 'yaml'
+
+module SSHScan
+  module ConfigLoader
+    class Yaml
+      def initialize(config, class_name = nil, mode = 'file', logger = nil)
```

Comment:
I think this is a good idea, to fail fast and hard if we get an exception in the config.